src/game.ts

```diff
@@ -0,0 +1,147 @@
+import { GameState } from "./gamestate";
+import { displayGameState } from "./render";
+import { createInterface } from "readline";
+
+const UTF16_CODE_OF_LETTER_A = 65;
+
+export interface Coordinates {
+  x: number;
+  y: number;
+}
+
+export async function runGameLoop(initialGameState: GameState) {
+  displayGameState(initialGameState);
+  let currentState = initialGameState;
+  let continueGame = true;
+  while (continueGame) {
+    const userInput = await askCellCoordinatesToUser();
+    let isUserCoordinatesAreValid = false;
+    while (!isUserCoordinatesAreValid) {
+      try {
+        const boardCoordinates = getBoardCoordinatesFromUserInput(userInput);
+        currentState = updateGameState(currentState, boardCoordinates);
+        isUserCoordinatesAreValid = true;
+      } catch (error) {
+        console.error(error.message);
+      }
+    }
+    displayGameState(currentState);
+  }
+}
+
+function askCellCoordinatesToUser(): Promise<string> {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+  return new Promise((resolve) => {
+    readline.question(
+      "Where would you like to put you stone ?",
+      (cellCoordinates) => {
+        readline.close();
+        resolve(cellCoordinates);
+      }
+    );
+  });
+}
+
+function getBoardCoordinatesFromUserInput(
+  userCoordinates: string
+): Coordinates {
+  let x: number, y: number;
+  if (userCoordinates.length < 2) {
+    throw new Error(
+      `Given coordinates is not compatible. Must be something like "H4"`
+    );
+  }
+  if (/[A-Z]/.test(userCoordinates[0])) {
+    x = userCoordinates[0].charCodeAt(0) - UTF16_CODE_OF_LETTER_A;
+  } else {
+    throw new Error(`Given character is not included in the range [A,Z]`);
+  }
+  y = parseInt(userCoordinates.slice(1, userCoordinates.length));
+  if (y >= 0 && y <= 25) {
+    return { x: x, y: y };
```

Comment:
Is it required all of these errors? If not I will be easier to check only a regexp and then get the value of x an y.